src/main/java/org/apache/flink/benchmark/RocksStateBackendBenchmark.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.benchmark;
+
+import org.apache.flink.benchmark.functions.IntLongApplications;
+import org.apache.flink.streaming.api.windowing.assigners.TumblingEventTimeWindows;
+import org.apache.flink.streaming.api.windowing.time.Time;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.OperationsPerInvocation;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.VerboseMode;
+
+import java.io.IOException;
+
+import static org.openjdk.jmh.annotations.Scope.Thread;
+
+@OperationsPerInvocation(value = RocksStateBackendBenchmark.RECORDS_PER_INVOCATION)
+public class RocksStateBackendBenchmark extends StateBackendBenchmarkBase {
+	public static final int RECORDS_PER_INVOCATION = 2_000_000;
+
+	public static void main(String[] args)
+			throws RunnerException {
+		Options options = new OptionsBuilder()
+				.verbosity(VerboseMode.NORMAL)
+				.include(".*" + RocksStateBackendBenchmark.class.getSimpleName() + ".*")
+				.build();
+
+		new Runner(options).run();
+	}
+
+	@Benchmark
+	public void stateBackends(RocksStateBackendContext context) throws Exception {
+		IntLongApplications.reduceWithWindow(context.source, TumblingEventTimeWindows.of(Time.seconds(10_000)));
+		context.execute();
+	}
+
+	@State(Thread)
+	public static class RocksStateBackendContext extends StateBackendContext {
+		@Param({"ROCKS", "ROCKS_INC"})
```

Comment:
should we also have `MEMORY` here so that we can compare?

Reply:
We don't need to, because `MEMORY` is already being tested by separate benchmark. Adding it here would just be redundant. 

Regardless of that, comparison between `ROCKS` and `MEMORY` is actually not that important, besides some general idea "how much slower it is" and some pure curiosity. For actual daily usage and productional value, you care only whether your recent changes slowed down or improved either `ROCKS` or `MEMORY`.

Reply:
Ops, and there are benchmarks for `MEMORY` `FS` etc, but there are in different class `MemoryStateBackendBenchmark`. I have split them, because they had to run on different number of records per invocation :( `ROCKS` was way too slow for `7 000 000` records and `MEMORY` is way too fast for `2 000 000` records :(

Reply:
ok, if we do not need the comparison, than this is ok